pandas/tests/frame/test_alter_axes.py

```diff
@@ -183,21 +198,20 @@ def test_set_index_pass_arrays_duplicate(self, frame_of_index_cols, drop,
         df.index.name = index_name
 
         keys = [box1(df['A']), box2(df['A'])]
+        result = df.set_index(keys, drop=drop, append=append)
 
-        # == gives ambiguous Boolean for Series
-        if drop and keys[0] is 'A' and keys[1] is 'A':
-            with tm.assert_raises_regex(KeyError, '.*'):
-                df.set_index(keys, drop=drop, append=append)
-        else:
-            result = df.set_index(keys, drop=drop, append=append)
-
-            # to test against already-tested behavior, we add sequentially,
-            # hence second append always True; must wrap in list, otherwise
-            # list-box will be illegal
-            expected = df.set_index([keys[0]], drop=drop, append=append)
-            expected = expected.set_index([keys[1]], drop=drop, append=True)
+        # if either box was iter, the content has been consumed; re-read it
+        keys = [box1(df['A']), box2(df['A'])]
 
-            tm.assert_frame_equal(result, expected)
+        # to test against already-tested behaviour, we add sequentially,
+        # hence second append always True; must wrap keys in list, otherwise
+        # box = list would be illegal; need to adapt drop for case that both
+        # keys are 'A' -- can't drop the same column twice
+        expected = df.set_index([keys[0]], drop=(False if (keys[0] is 'A'
```

Comment:
can you make the drop a separate statement, e.g.

```
drop=(False.....)
expected = df.set_index(....
```

Reply:
That's what I had before (with `first_drop`). I can't overwrite `drop` because it's needed for the second `set_index`.